src/plugins/data_source_management/server/saved_objects/data_source.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { SavedObjectsType } from 'opensearch-dashboards/server';
+
+export const dataSource: SavedObjectsType = {
+  name: 'data-source',
+  namespaceType: 'agnostic',
+  hidden: false,
+  management: {
+    icon: 'apps', // todo: pending ux #2034
+    defaultSearchField: 'title',
+    importableAndExportable: true,
+    getTitle(obj) {
+      return obj.attributes.title;
+    },
+    // todo: update getEditUrl & getInAppUrl after #2021
+  },
+  mappings: {
+    dynamic: false,
+    properties: {
+      title: {
+        type: 'text',
+      },
+      type: {
```

Comment:
What type could it be? "opensearch"?

Reply:
data-source

Reply:
Is it really needed? a field in mapping with one fixed value?

Reply:
Good question we should not need this. Removed